lib/main.dart

```diff
@@ -36,25 +40,74 @@ class V34 extends StatefulWidget {
 }
 
 class _V34State extends State<V34> {
+  Future<SharedPreferences> _preferences;
+
+  @override
+  void initState() {
+    super.initState();
+    _preferences = SharedPreferences.getInstance();
+  }
+
+  ThemeData _getTheme(AsyncSnapshot<SharedPreferences> snapshot) {
+    bool automatic = snapshot.data.getBool("automatic_dark_theme") ?? false;
+    bool dark = snapshot.data.getBool("dark_theme") ?? false;
+    DateTime now = DateTime.now();
+    if (automatic) {
+      if (now.hour >= 20 || now.hour < 8) return AppTheme.darkTheme();
```

Comment:
I have seen this test twice in this PR. You can probably factorize them in a function / method, to be sure that any change will be applied everywhere